routing/turns_tts_text.cpp

```diff
@@ -20,18 +20,23 @@ using namespace routing::turns::sound;
 
 template <class TIter> std::string DistToTextId(TIter begin, TIter end, uint32_t dist)
 {
-  using TValue = typename std::iterator_traits<TIter>::value_type;
-
-  TIter distToSound = lower_bound(begin, end, dist, [](TValue const & p1, uint32_t p2)
-                      {
-                        return p1.first < p2;
-                      });
-  if (distToSound == end)
+  TIter it = lower_bound(begin, end, dist, [](auto const & p1, uint32_t p2) { return p1.first < p2; });
+  if (it == end)
   {
     ASSERT(false, ("notification.m_distanceUnits is not correct."));
-    return std::string{};
+    return {};
   }
-  return distToSound->second;
+
+  if (it != begin)
+  {
+    // Rounding like 130 -> 100; 135 -> 200 is better than upper bound, IMO.
```

Comment:
#7147 can help. What's going on here in case 135 -> 200? When TTS starts, driver will be at 100 meters already, will he hear 200?

Reply:
Before:
101 -> 200
102 -> 200
...
199 -> 20

After:
101 -> 100
102 -> 100
...
133 -> 100
134 -> 200
...
199 -> 200

Reply:
Why 135 is chosen instead of 150 or 175? What is the logic?

Reply:
What is your opinion on how is better?
We can pronounce 100 or 200 meters. What to pronounce, when the next turn is on 120, 135, 150, 175 meters?

Reply:
I'm trying to understand when it happens and why. Do you have the understanding? A good TTS should alert users only when the distance is at `200m + (curr speed * time_to_talk_200m_announcement)` and at `100m + (curr speed * time_to_talk_100m_announcement)` and being silent at any other intermediate values.

Isn't it confusing to say to a driver (or a pedestrian, or cyclist) that there are 200m left when in reality it is 135m? Or to say that 100m left when in reality it is 130m?

Reply:
- When start routing and the next turn within 400 meters, like: "turn left and then in xxx meters turn right"
- When exiting roundabout, like: "In XXX meters take Y exit"

Reply:
We need to add separate strings here like "In %$1d meters take %$2d exit" instead of that custom patching. @zyphlar can you please create a separate PR for that?

Reply:
- AFAIK, none of the navigators says meters for roundabout exits
- I suspect that pronouncing every possible numeric value will be ugly
- What to do with "... then ..." The same?

Reply:
Rounding to the nearest 10 meters should be appropriate IMO. When you start nav, you're not moving. When you're in a circle, the distance is usually not that great.

Confusing with 200m instead of 135m doesn't sound right. And I agree that it's better than now (200m instead of 101m).

Created an issue https://github.com/organicmaps/organicmaps/issues/7175